config/initializers/aws_cloudwatch.rb

```diff
@@ -1,60 +1,23 @@
 Rails.logger.debug "Running initializer in #{__FILE__}"
 
 def find_log_stream_name
-  log_stream = nil
-  next_token = nil
-  begin
-    Timeout.timeout(15) do
-      log_group = ENV.fetch('TARGET_GROUP_NAME', nil)
-
-      if log_group.nil?
-        Rails.logger.info 'TARGET_GROUP_NAME not set, bailing out'
-        return nil
-      end
-
-      begin
-        task_meta = Net::HTTP.get( URI( "#{ENV['ECS_CONTAINER_METADATA_URI_V4']}/task" ) )
-        task_arn  = JSON.parse(task_meta)["TaskARN"]
-        task_id   = task_arn.split('/').last
-      rescue => e
-        Rails.logger.error 'Something broke when querying for the ECS task id.'
-        Rails.logger.error e.message
-      end
-
-      logs ||= Aws::CloudWatchLogs::Client.new
-      for i in 1..100 do # Limit to 100 requests so we're not endlessly searching.
-        begin
-          sleep(2) if i % 5 == 0
-          response = logs.describe_log_streams({
-            log_group_name: log_group,
-            order_by: 'LastEventTime',
-            descending: true,
-            next_token: next_token,
-          })
-        rescue Aws::CloudWatchLogs::Errors::ThrottlingException, Timeout::Error  => e
-          Rails.logger.error 'Throttling exception encountered when searching for log stream.'
-          return nil
-        end
+  log_group = ENV.fetch('TARGET_GROUP_NAME', nil)
 
-        log_streams = response.log_streams
-        log_stream = log_streams.find { |s| s.log_stream_name.include?(task_id) }
-        if log_stream.nil?
-          next_token = response.next_token
-          next
-        else
-          return log_stream.log_stream_name
-        end
-      end
-    end
-  rescue Timeout::Error  => e
-    Rails.logger.error 'Searching for the log stream took too long.'
+  if log_group.nil?
+    Rails.logger.info 'TARGET_GROUP_NAME not set, bailing out'
     return nil
   end
 
-  if log_stream.nil?
-    Rails.logger.error 'Log stream not found within 100 requests.'
-    return nil
+  begin
+    task_meta = Net::HTTP.get( URI( "#{ENV['ECS_CONTAINER_METADATA_URI_V4']}/task" ) )
+    task_arn  = JSON.parse(task_meta)["TaskARN"]
+    task_id   = task_arn.split('/').last
+  rescue => e
```

Comment:
Maybe we should rescue StandardError instead? Should this return nil or something?